modules/UI/UI.js

```diff
@@ -824,9 +824,22 @@ UI.emitEvent = function (type, options) {
 };
 
 UI.clickOnVideo = function (videoNumber) {
-    var remoteVideos = $(".videocontainer:not(#mixedstream)");
-    if (remoteVideos.length > videoNumber) {
-        remoteVideos[videoNumber].click();
+    let videos = $("#remoteVideos .videocontainer:not(#mixedstream)").toArray();
```

Comment:
My proposal for optimization of the method is:
```
UI.clickOnVideo = function (videoNumber) {
    let videos = $("#remoteVideos .videocontainer:not(#mixedstream)");
    let videosLength = videos.length;
    if(videosLength <= videoNumber) {
        return;
    }
    videos[(videoNumber === 0) ? 0 : videosLength - videoNumber].click();
}
```

The elements are ordered in the jquery selector like they are ordered in the document and since we know that the most left element is actually the last video element in the document and should be matched to number 1 and number 2 should correspond to the previous element in the document, etc. This means that the index of the element in the document order is `the number of elements - the pressed number`. The only exception is the local video which should correspond always to 0.

This way we are saving a lot of operations and construction of 2 arrays and 1 map.